src/pages/SingleCell/HomePage.tsx

```diff
@@ -0,0 +1,1168 @@
+            chartInfo: {
+                name: '',
+                description: '',
+                profileType: '',
+                genericAssayType: '',
+                dataType: '',
+                genericAssayEntityId: '',
+                patientLevel: false,
+            },
+            selectedEntity: null,
+            selectedValue: null, // Initialize selectedValue as null
+            dataBins: null, // Initialize dataBins as null
+            chartType: null, // Initialize chartType as null
+            pieChartData: [],
+            tooltipEnabled: false,
+            downloadSvg: false,
+            downloadPdf: false,
+            downloadOption: '',
+            BarDownloadData: [],
+            stackEntity: '',
+            studyIdToStudy: '',
+            hoveredSampleId: [],
+            currentTooltipData: [],
+            map: {},
+            dynamicWidth: 0,
+            increaseCount: 0,
+            decreaseCount: 0,
+            resizeEnabled: false,
+            isHorizontal: false,
+            isVisible: false,
+            tooltipHovered: false,
+            selectedSamples: [],
+            dropdownOptions: [],
+            isReverse: false,
+            initialWidth: 0,
+            heading: '',
+            isHovered: false,
+            hoveredSliceIndex: 0,
+            profileTypeBin: '',
+            stableIdBin: '',
+            databinState: [],
+        };
+    }
+    async fetchGenericAssayData(
+        selectedValue: string,
+        names: string[],
+        sampleId: string[]
+    ) {
+        const { store } = this.props;
+        const params = {
+            molecularProfileId: selectedValue,
+            genericAssayFilter: {
+                genericAssayStableIds: names,
+                sampleIds: sampleId,
+            } as GenericAssayFilter,
+        };
+
+        try {
+            const resp = await client.fetchGenericAssayDataInMolecularProfileUsingPOST(
+                params
+            );
+            return resp;
+        } catch (error) {
+            console.error('Error fetching generic assay data', error);
+        }
+    }
+
+    @autobind
+    handleTooltipCheckboxChange(event: React.ChangeEvent<HTMLInputElement>) {
+        singleCellStore.setTooltipEnabled(event.target.checked);
+        this.setState({ tooltipEnabled: event.target.checked });
+    }
+    @autobind
+    handleReverseChange(event: React.ChangeEvent<HTMLInputElement>) {
+        singleCellStore.setIsReverse(event.target.checked);
+        this.setState({ isReverse: event.target.checked });
+    }
+    async fetchDataBins(genericAssayEntityId: string, profileType: string) {
+        let temp = this.props.store.genericAssayProfileOptionsByType.result;
+        let id = temp[profileType];
+        let tempstudyId = id[0].value;
+        singleCellStore.setHeading(profileType);
+        singleCellStore.setStableIdBin(genericAssayEntityId);
+        singleCellStore.setProfileTypeBin(tempstudyId);

```

Comment:
instead of first defining `temp`, `id`, and `tempstudyId`, can  you not directly set the study id:
`singleCellStore.setProfileTypeBin(this.props.store.genericAssayProfileOptionsByType.result[profileType].id[0].value)`